packages/coinjoin/src/client/round/connectionConfirmation.ts

```diff
@@ -63,20 +77,13 @@ const confirmInput = async (
         )
         .catch(error => {
             log(`Confirmation failed ~~${input.outpoint}~~ in ~~${round.id}~~. Reason: ${error}`);
-            // catch specific error
-            if (
-                error.message ===
-                coordinator.WabiSabiProtocolErrorCode.AliceAlreadyConfirmedConnection
-            ) {
-                // NOTE: possible race condition between round phases:
-                // 1. suite is constantly confirming connection while in phase: 0 (see confirmationInterval below)
-                // 2. coordinator changes round phase to 1 before the deadline estimated by suite, suite doesn't know about phase change yet and is still in confirmation interval process
-                // 3. confirmation from interval is successfully sent as confirmation in phase: 1
-                // 5. suite receives phase change from coordinator and trying to process confirmation in phase: 1
-                // and here we are... ignoring it, if input doesn't have confirmedAmountCredentials it will fail in next phase
-            } else {
-                throw error;
+            // if error is not WabiSabiProtocolErrorCode like AliceNotFound, AliceAlreadyConfirmedConnection or other protocol violation
+            if (!Object.keys(coordinator.WabiSabiProtocolErrorCode).includes(error.message)) {
+                // then store failed params for recovering (see description above)
+                input.setConfirmationParams(zeroAmountCredentials, zeroVsizeCredentials);
```

Comment:
Does it matter that `AliceAlreadyConfirmedConnection` didn't throw before but now it does?

Reply:
previous implementation was wrong, it was ignoring this error and moving on to the next phase. in the next phase input didnt have `confirmationData` (credentials) so no outputs was created from it.

with new implementation "protocol violation" errors means: "stop here there is nothing you can do" and any other error (like timeout, abort etc ) means: "store used params (zero credentials) for future use"

those exceptions usually happens between phase 0 and 1 when:
- round is in phase 0 and request deadline or timeout is triggered
- round is switching from phase 0 to phase 1 before expected deadline. suite assumes that registr. phase will end in 3 minutes but it actually ends in 2 minutes while connectionConfirmationInterval is running with delayed request

Reply:
actually im putting a follow up for that, because i think it might be also useful in `credentials-issuance` request (similar case)